Source/panels/charpanel.cpp

```diff
@@ -199,16 +199,19 @@ void DrawPanelField(const Surface &out, Point pos, int len)
 
 void DrawShadowString(const Surface &out, const PanelEntry &entry)
 {
-	if (entry.label == "")
+	if (entry.label.empty())
 		return;
 
-	std::string text_tmp = _(entry.label.c_str());
-	char buffer[64];
-	int spacing = 0;
-	strcpy(buffer, text_tmp.c_str());
-	if (entry.labelLength > 0)
-		WordWrapString(buffer, entry.labelLength, GameFont12, spacing);
-	std::string text(buffer);
+	constexpr int Spacing = 0;
+	const std::string &textStr = LanguageTranslate(entry.label.c_str());
+	string_view text;
+	std::string wrapped;
+	if (entry.labelLength > 0) {
+		wrapped = WordWrapString(textStr, entry.labelLength, GameFont12, Spacing);
+		text = wrapped;
+	} else {
+		text = textStr;

```

Comment:
I don't think labels with 0 length is allowed /used

Reply:
From the code it seems length 0 means "Do not wrap". Sounds like this change should be made separately though.

Reply:
Right, I can see now that you added the `else` as a substitution for the `strcpy`, not as a new new condition.